### PR TITLE
 [JENKINS-57616] JCasC compatibility

### PR DESCRIPTION
**Pull request by rsandell**
https://issues.jenkins-ci.org/browse/JENKINS-57616

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
     <properties>
         <jenkins.version>2.60.3</jenkins.version>
         <java.level>8</java.level>
+        <jcasc.version>1.27</jcasc.version>
```
Comment (varyvol) [inline pom.xml]: Why not latest version?

File: src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/SecureRequesterImplTest.java [hunk 2]
```diff
@@ -42,7 +42,10 @@
     @Test public void authorizing() throws Exception {
         assertJSONP(null, 403);
         assertJSONP("http://apache.org/", 403);
-        Whitelist.get().configure(null, new JSONObject().accumulate("allowNoReferer", true).accumulate("domains", "apache.org jenkins-ci.org"));
+        final Whitelist whitelist = Whitelist.get();
```
Comment (varyvol) [inline src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/SecureRequesterImplTest.java]: Why is this change necessary?

File: src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/CasCTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+public class CasCTest {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @Issue("JENKINS-57616")
+    @ConfiguredWithCode("CasCTest.yml")
+    public void configureWhitelist() throws Exception {
```
Comment (varyvol) [inline src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/CasCTest.java]: You can extend from a test class yo make this cleaner (I think). See https://github.com/jenkinsci/ant-plugin/pull/36 as en example.

File: pom.xml (hunk 1 above)
Comment (oleg-nenashev) [reply pom.xml]: It rolls fast.

```suggestion
        <jcasc.version>1.29</jcasc.version>
```

File: src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/SecureRequesterImplTest.java (hunk 2 above)
Comment (rsandell) [reply src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/SecureRequesterImplTest.java]: The old code sent in a json object but a `null` `StaplerRequest` the new code needs it to do the binding.

File: src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/CasCTest.java (hunk 3 above)
Comment (rsandell) [reply src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/CasCTest.java]: I'm not sure what you mean, what makes the ant example cleaner?

Comment (varyvol) [reply src/test/java/org/jenkinsci/plugins/secure_requester_whitelist/CasCTest.java]: Basically you save the rule, the @test, the @configureWithCode.

File: pom.xml (hunk 1 above)
Comment (rsandell) [reply pom.xml]: I knew that 1.27 had core 2.60.x and didn't bother to investigate if newer versions existed and had the same core dependency :)

File: src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java [hunk 4]
```diff
@@ -30,19 +30,25 @@
...
 import jenkins.model.Jenkins;
 import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
-@Extension public class Whitelist extends GlobalConfiguration {
+@Extension
+@Symbol("secure_requester_whitelist")
```
Comment (fcojfernandez) [inline src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java]: Why did you have to add the Symbol now? is it needed for JCasC, good practice, other reason?

Comment (Wadeck) [reply src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java]: The symbol could be used by JCasC as a facilitator, not strictly required but largely encouraged I would say :)

Comment (fcojfernandez) [reply src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java]: :+1: Good to know!

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java]:
```suggestion
@Symbol("secureRequesterWhitelist")
```

Just to follow the standard naming pattern in JCASc

Comment (rsandell) [reply src/main/java/org/jenkinsci/plugins/secure_requester_whitelist/Whitelist.java]: Yes, it is needed for JCasC. I haven't tried without so I don't know what JCasC does when there is no symbol. Pipeline tries to guess based on class name and in that case it would be `Whitelist` which is not very descriptive for the plugin.

**Comment by rsandell** [comment]
@oleg-nenashev Done

**Comment by rsandell** [comment]
@Wadeck @daniel-beck ping?

**Comment by Wadeck** [comment]
@oleg-nenashev is the new name fine with you? So that you can remove your request change.

**Comment by Wadeck** [comment]
Will create a release with this in the afternoon
